include/seadsa/SpecGraphInfo.hh

```diff
@@ -0,0 +1,56 @@
+/**
+   A pass to replace bottom-up PTA of identified functions with a pre-generated
+   spec
+ */
+#pragma once
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Analysis/TargetLibraryInfo.h"
+#include "llvm/Pass.h"
+
+#include "seadsa/AllocWrapInfo.hh"
+#include "seadsa/Local.hh"
+
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+#include "llvm/IR/Value.h"
+#include <unordered_map>
+
+namespace llvm {
+class Function;
+class Module;
+class TargetLibraryInfoWrapperPass;
+class Function;
+} // namespace llvm
+
+namespace seadsa {
+class LocalAnalysis;
+class AllocWrapInfo;
+
+class SpecGraphInfo : public llvm::ImmutablePass {
+
+protected:
+  mutable AllocWrapInfo *m_awi;
+  mutable llvm::TargetLibraryInfoWrapperPass *m_tliWrapper;
+  mutable std::unordered_map<std::string, llvm::Function &> m_funcs;
```

Comment:
maybe better `const llvm::Function*` rather than `llvm::Function&`?

Reply:
I can do this, but I'm not sure what the reasoning behind this is?

Reply:
can't have it const because of the call [here](https://github.com/seahorn/sea-dsa/blob/02a7dfb09068a27d1261fbc115a98dec0a915cb5/lib/seadsa/DsaBottomUp.cc#L148), but I agree that having it as pointer type is a better choice. Change here: 2245707